### PR TITLE
Document warning about JDBC catalog tables in Iceberg

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Fixes https://github.com/trinodb/trino/issues/17744

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

File: docs/src/main/sphinx/connector/metastores.md [hunk 1]
```diff
@@ -370,6 +370,10 @@ directory.
 The JDBC catalog may have compatibility issues if Iceberg introduces breaking
 changes in the future. Consider the {ref}`REST catalog
 <iceberg-rest-catalog>` as an alternative solution.
+
+The JDBC catalog requires the existing metadata tables.
```
Comment (hashhar) [inline docs/src/main/sphinx/connector/metastores.md:374]:
```suggestion
The JDBC catalog requires the metadata tables to already exist. 
```

**Review by hashhar — APPROVED** [review]

**Comment by mosabua** [comment]
fyi @bitsondatadev .. let us know when there is better docs for creating the tables than the source code.

**Comment by tschaub** [comment]
Has anyone figured out how to use the the linked [JdbcUtil.java](https://github.com/apache/iceberg/blob/apache-iceberg-1.5.0/core/src/main/java/org/apache/iceberg/jdbc/JdbcUtil.java) to initialize tables for a JDBC-based iceberg metastore?

**Comment by luanmm** [comment]
The issue #17744 states the need to initialize Iceberg catalog, but Trino should be passing the "initializeCatalogTables" option to Iceberg do it in the first run.

Iceberg would do it "automatically" if configured to. It is not something that users should do themselves.

It is pointless to link a source code in docs for most of the users. Maybe making a PR passing the parameter to Iceberg would be even simpler than writing this as a requirement in the docs.

**Pull request merged**